### PR TITLE
feat: [OpenAPI] Improved OneOf Support

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

Follow up to #647, see [this comment](https://github.com/SAP/cloud-sdk-java/pull/647#discussion_r1860833002) 

### Feature scope:

- [x] `oneOf` deserialization support without discriminators
- [x] remove `@Type` property from serialization
- [x] serialization test
- [x] general template improvements
- [x] easier template debugging via property

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Documentation updated~
- [x] ~Release notes updated~

### CODE REVIEW

File: datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminator.java [hunk 1]
```diff
@@ -22,13 +22,12 @@
...
-@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "sodaType", visible = true )
+@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, property = "sodaType", visible = true )
 @JsonSubTypes( {
     @JsonSubTypes.Type( value = Cola.class, name = "Cola" ),
     @JsonSubTypes.Type( value = Fanta.class, name = "Fanta" ), } )
 
 public interface OneOfWithDiscriminator
 {
-    public String getSodaType();
+    String getSodaType();
```
Comment (MatKuhr) [inline datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminator.java:32]: public is redundant for interfaces

**Review by CharlesDuboisSAP — COMMENTED** [review]
Can we see tCan you create a PR in the AI SDK to see the removed mixIn

File: datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOf.java [hunk 2]
```diff
@@ -16,13 +16,14 @@
...
+import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 
 /**
  * OneOf
  */
-
-@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "", visible = true )
+@JsonTypeInfo( use = JsonTypeInfo.Id.DEDUCTION )
+@JsonSubTypes( { @JsonSubTypes.Type( value = Cola.class ), @JsonSubTypes.Type( value = Fanta.class ), } )
```
Comment (newtork) [inline datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOf.java:26]: (Question)

_Deduction_ is only applied when there is **NO** discriminator, correct?

If there is a discriminator, then the property based annotation is used instead(?)

File: datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminatorAndMapping.java [hunk 3]
```diff
@@ -22,13 +22,14 @@
...
  * OneOfWithDiscriminatorAndMapping
  */
-
-@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "sodaType", visible = true )
+@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, property = "sodaType", visible = true )
 @JsonSubTypes( {
+    @JsonSubTypes.Type( value = Cola.class, name = "cool_cola" ),
+    @JsonSubTypes.Type( value = Fanta.class, name = "fancy_fanta" ),
     @JsonSubTypes.Type( value = Cola.class, name = "Cola" ),
     @JsonSubTypes.Type( value = Fanta.class, name = "Fanta" ), } )
```
Comment (newtork) [inline datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminatorAndMapping.java:30]: This looks strange to me, why are cola/fanta referenced twice?

Comment (CharlesDuboisSAP) [reply datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminatorAndMapping.java:30]: `cool_cola` the custom mapping and `Cola` the default mapping

File: datamodel/openapi/openapi-generator/src/main/java/com/sap/cloud/sdk/datamodel/openapi/generator/GenerationConfigurationConverter.java [hunk 4]
```diff
@@ -104,6 +104,9 @@ private static void setGlobalSettings( @Nonnull final GenerationConfiguration co
         if( configuration.isGenerateModels() ) {
             GlobalSettings.setProperty(CodegenConstants.MODELS, "");
         }
+        if( configuration.isDebugModels() ) {
+            GlobalSettings.setProperty("debugModels", "true");
+        }
```
Comment (newtork) [inline datamodel/openapi/openapi-generator/src/main/java/com/sap/cloud/sdk/datamodel/openapi/generator/GenerationConfigurationConverter.java:109]: (Question)

Will you remove this?

**Comment by MatKuhr** [comment]
> Can you create a PR in the AI SDK to see the removed mixIn

will do once #652 is merged to not have too many different generator changes applied in parallel

File: datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminatorAndMapping.java (hunk 3 above)
Comment (MatKuhr) [reply datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOfWithDiscriminatorAndMapping.java:30]: The OSS generator always seems to apply a default mapping, I don't know why

File: datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOf.java (hunk 2 above)
Comment (MatKuhr) [reply datamodel/openapi/openapi-api-sample/src/main/java/com/sap/cloud/sdk/datamodel/openapi/sample/model/OneOf.java:26]: correct 👍🏼

**Comment by CharlesDuboisSAP** [comment]
> > Can you create a PR in the AI SDK to see the removed mixIn
> 
> will do once #652 is merged to not have too many different generator changes applied in parallel

Done, let me know when the AI SDK PR is created

**Comment by MatKuhr** [comment]
[**Done**](https://github.com/SAP/ai-sdk-java/pull/235)

**Review by CharlesDuboisSAP — APPROVED** [review]

**Pull request merged**